local_planner/include/local_planner/local_planner_visualization.h

```diff
@@ -0,0 +1,164 @@
+#ifndef LOCAL_PLANNER_VISUALIZATION_H
+#define LOCAL_PLANNER_VISUALIZATION_H
+
+#include "local_planner/local_planner.h"
+
+#include <pcl/point_cloud.h>
+#include <pcl_ros/point_cloud.h>
+#include <ros/ros.h>
+#include <Eigen/Dense>
+#include <vector>
+
+namespace avoidance {
+
+class LocalPlannerVisualization {
+ public:
+  /**
+  * @brief      initializes all subscribers used for local planner visualization
+  **/
+  void initializeSubscribers(ros::NodeHandle& nh);
+
+  /**
+  * @brief       Main function which calls functions to visualize all planner
+  *output ready at the end of one planner iteration
+  * @params[in] planner, reference to the planner
+  * @params[in] newest_waypoint_position, last caluclated waypoint (smoothed)
+  * @params[in] newest_adapted_waypoint_position, last caluclated waypoint
+  *(non-smoothed)
+  **/
+  void visualizePlannerData(
+      LocalPlanner& planner,
```

Comment:
This is somewhat questionable - If the LocalPlanner owns a LocalPlannerVisualization, which has a reference to a LocalPlanner, you create a cyclic dependency. It's possible - but not really good OO architecture I think

Reply:
Agreed, circular dependencies are bad. Either the local planner should pass the data into the visualization (maybe with a struct?), or the local planner shouldn't own the copy of the visualization (maybe it lives in the node instead.

Reply:
In my opinion this is not a cyclic dependency: local_planner node owns a local_planner and a visualizer. After the local_planner has done a run, the local_planner_NODE passes the reference to the local_planner into the visualizer. I do not see a circle here